tests/features/steps/open-files.py

```diff
@@ -0,0 +1,300 @@
+                    files=files,
+                    verify=False)
+
+    context.response= r
+    context.status_code=r.status_code
+    print(context.status_code,context.response)
+
+@when('The consumer publishes with empty token')
+def step_imp(context):
+    payload= (
+           ("id",res[0]
+            ),
+    ('token', ''),
+
+)
+    files = {
+    'file': ('sample.txt', open('sample.txt', 'rb')),
+    'metadata': ('meta.json', open('meta.json', 'rb')),
+    }
+
+
+
+    r=requests.post(url=VERMILLION_URL + PUBLISH_ENDPOINT,
+                    data=payload,
+                    files=files,
+                    verify=False)
+
+    context.response= r
+    context.status_code=r.status_code
+    print(context.status_code,context.response)
+
+@when('The consumer publishes by removing file form parameter')
+def step_imp(context):
+    payload= (
+           ("id",res[0]
+            ),
+    ('token', tokens["master"]),
+
+)
+    files = {
+    #'file': ('sample.txt', open('sample.txt', 'rb')),
+    'metadata': ('meta.json', open('meta.json', 'rb')),
+    }
+
+
+
+    r=requests.post(url=VERMILLION_URL + PUBLISH_ENDPOINT,
+                    data=payload,
+                    files=files,
+                    verify=False)
+
+    context.response= r
+    context.status_code=r.status_code
+    print(context.status_code,context.response)
+
+@when('The consumer publishes by removing metadata form parameter')
+def step_imp(context):
+    payload= (
+           ("id",res[0]
+            ),
+    ('token', tokens["master"]),
+
+)
+    files = {
+    'file': ('sample.txt', open('sample.txt', 'rb')),
+    #'metadata': ('meta.json', open('meta.json', 'rb')),
+    }
+
+
+
+    r=requests.post(url=VERMILLION_URL + PUBLISH_ENDPOINT,
+                    data=payload,
+                    files=files,
+                    verify=False)
+
+    context.response= r
+    context.status_code=r.status_code
+    print(context.status_code,context.response)
+
+
+@when('The consumer publishes by using extraneous form parameter')
+def step_imp(context):
+    payload= (
+           ("id",res[0]
+            ),
+    ('token', tokens["master"]),
+
+)
+    files = {
+    'abc': ('samplecsv.csv', open('samplecsv.csv', 'rb')),
+    'efg': ('samplepdf.pdf', open('samplepdf.pdf', 'rb')),
+    }
+
+    
+    fil=glob.glob('../api-server/file-uploads/*')
+    for f in fil:
+        os.remove(f)
+    
+
+    r=requests.post(url=VERMILLION_URL + PUBLISH_ENDPOINT,
+                    data=payload,
+                    files=files,
+                    verify=False)
+
+    context.response= r
+    context.status_code=r.status_code
+    print(context.status_code,context.response)
+
+@then('The uploaded files are deleted')
+def step_imp(context): 
+    DIR = '../api-server/file-uploads'
+    number_of_files = len([name for name in os.listdir(DIR) if os.path.isfile(os.path.join(DIR, name))])
+    counter = 0
+    while( number_of_files > 0 and counter < 60):
```

Comment:
Kindly add a comment as to why this is necessary @vasanth096

Reply:
Added comments

Reply:
Have you added them elsewhere? I am not able to see them

Reply:
@then is added in then.py

Reply:
Ok